drivers/gpio/gpio_intel.c

```diff
@@ -435,19 +440,19 @@ static int port_set_raw(const struct device *dev, uint32_t mask,
 	while (mask != 0) {
 		pin = find_lsb_set(mask) - 1;
 
-		if (pin > cfg->num_pins) {
+		if (pin >= cfg->num_pins) {
```

Comment:
Can you comment on this change? Was it a bug in the APL driver, or did you update the `ngpios` property in the APL DTS so that this still does the same thing on APL?

Reply:
...and if it was a bug, it should probably be split off into its own commit, since it's not about adding EHL support per-se.

Reply:
It was a bug.  I will split off the commit, would you suggest adding it on a different PR?

Reply:
I think you can keep it in this same PR, since it all relates to the same driver